xchange-kraken/src/main/java/org/knowm/xchange/kraken/service/KrakenAccountService.java

```diff
@@ -39,9 +48,50 @@ public String requestDepositAddress(Currency currency, String... args) throws IO
     return KrakenAdapters.adaptKrakenDepositAddress(depositAddresses);
   }
 
-  // public WithdrawInfo[] getWithdrawInfo(String assetPairs, String assets,
-  // String key, BigDecimal amount)
-  // throws IOException {
-  // return getWithdrawInfo(assetPairs, assets, key, amount);
-  // }
+  @Override
+  public FundsInfo getFundsInfo(TradeHistoryParams params) throws ExchangeException, NotAvailableFromExchangeException, NotYetImplementedForExchangeException, IOException{
+    KrakenFundsInfoHistoryParams histParams = (KrakenFundsInfoHistoryParams) params;
+    String startTime = null;
+    if (histParams.getStartTime() != null){
+      startTime = String.valueOf(DateUtils.toUnixTime(histParams.getStartTime()));
+    }
+    String endTime = null;
+    if (histParams.getEndTime() != null){
+      endTime = String.valueOf(DateUtils.toUnixTime(histParams.getEndTime()));
+    }
+    String offset = null;
+    if (histParams.getOffset() != null){
+      offset = String.valueOf(histParams.getOffset());
+    }
+
+    return KrakenAdapters.adaptFundsInfo(getKrakenLedgerInfo(null, startTime, endTime, offset, histParams.assets));
+  }
+
+
+  public static class KrakenFundsInfoHistoryParams extends DefaultTradeHistoryParamsTimeSpan implements TradeHistoryParamOffset{
```

Comment:
I suggest implementing `TradeHistoryParamCurrency` (see [this comment](https://github.com/timmolter/XChange/pull/1406/commits/88f05bf33a91800b5df6c026222206a4495b9016#r107113167)) here too; the setter would set `assets` to the appropriate single-value array, and the getter can throw an `UnsupportedOperationException` as it won't be called anyway.

This way users will be able to use the interface in a completely exchange-agnostic way, but will still have the ability (with a few lines of Kraken-specific code) to use Kraken's multi-currency funds info request through the generic interface.